src/sim-host/socket.js

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
This file is not just a UI related module. It is part of the sim-host client side implementation but it is critical for the main communication between app-host and server. Is the sim-host side implementation of the protocol for simulation.
If we enable developers to change this, they must know the protocol, and maybe they end up doing the same than here, when they only wants to provide a customised UI.
So, maybe we can do this:
- provide the default implementation but from the server side that is only for sim-host, so we should move `socket.js` to some folder in the server
- Let them load it on `sim-host.js` module, by default our implementation will load the socket implementation from our server, instad of doing this https://github.com/Microsoft/cordova-simulate/pull/11/files#diff-c1aa19020ad6303656f2fbf0b9d79df4R6, we can do `var socket = require('socket')`. We can also rename the script name to be more representative, since it handles the sim-host side of the protocol, it could be protocol.js, under some sim-host folder in the server.
- if the developers of the new sim-host UI needs to tweak a little the protocol for them, wait for something in the UI before connecting or something, then they can provide this file and must follow the protocol defined for the communication, then they should do `var protocol = require('my-modified-protocol')` instead of the default one.

I have my doubts about the last point: let developers modified the sim-host side protocol. If modifying the protocol is a requirement for other developers, maybe the recommendation for them is to clone the whole repo and make the modification as needed in the protocol. What do they need that they provide a new protocol? If it is refreshing sim-host side in a different way than just calling `document.location.reload(true);`, or handling the failures differently, by not logging the error but doing something else, then we can provide some optional configuration to the initialization call that enable developers to customize some actions.